src/js/components/conversation.jsx

```diff
@@ -54,34 +66,146 @@ export class ConversationComponent extends Component {
 
     };
 
+    onScroll = () => {
+        const {dispatch, shouldScrollToBottom} = this.props;
+
+        // If top of Conversation component is reached, we need to fetch older messages
+        const node = findDOMNode(this);
+        if (node.scrollTop === 0) {
+            this.fetchHistory();
+        } else if (shouldScrollToBottom) {
+            // Once we've started scrolling, we don't want the default behavior to force the scroll to the bottom afterwards
+            dispatch(setShouldScrollToBottom(false));
+        }
+    };
+
+    fetchHistory = () => {
+        const {dispatch, hasMoreMessages, isFetchingMoreMessages, messages} = this.props;
+
+        const node = findDOMNode(this);
+        if (hasMoreMessages && !isFetchingMoreMessages) {
```

Comment:
I think instead of making this if statement check inside fetchHistory, it should be `onScroll` who decides whether or not it should call `fetchHistory`.

There are already such checks deciding if fetchHistory gets called in the `render()` method below.